### PR TITLE
Add varint encoding decoding

### PR DESCRIPTION
**Pull request by i1i1**
It is a bugfix for public key decoding and encoding. Multihash uses Variable length uints. This commits adds varuint.

Signed-off-by: i1i1 <vanyarybin1@live.ru>

### CODE REVIEW

**Review by humb1t — APPROVED** [review]

File: iroha_crypto/src/varint.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Debug)]
+pub struct VarUint {
+    payload: Vec<u8>,
+}
```
Comment (eadventurous) [inline iroha_crypto/src/varint.rs:4]: Please add doc comments

File: iroha_crypto/src/varint.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+);
+
+from_uint!(u8);
+from_uint!(u16);
+from_uint!(u32);
+from_uint!(u64);
+from_uint!(u128);
+
+impl VarUint {
+    pub fn new(bytes: impl AsRef<[u8]>) -> Result<Self, String> {
```
Comment (eadventurous) [inline iroha_crypto/src/varint.rs:74]: Also here

File: iroha_crypto/src/multihash.rs [hunk 3]
```diff
@@ -79,11 +84,18 @@ pub struct Multihash {
...
+        let idx = bytes
+            .iter()
+            .enumerate()
+            .find(|&(_, &byte)| (byte & 0b1000_0000) == 0)
+            .ok_or_else(|| "Last byte should be less than 128".to_owned())?
+            .0;
+        let (digest_function, bytes) = bytes.split_at(idx + 1);
+        let mut bytes = bytes.iter().copied();
+
+        let digest_function: u64 = super::varint::VarUint::new(digest_function)?.try_into()?;
```
Comment (eadventurous) [inline iroha_crypto/src/multihash.rs:96]: Import `super::varint::VarUint`

File: iroha_crypto/src/varint.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Debug)]
+pub struct VarUint {
```
Comment (eadventurous) [inline iroha_crypto/src/varint.rs:2]: Please add module docs with a link to the spec https://github.com/multiformats/unsigned-varint

**Review by eadventurous — APPROVED** [review]
Everything else seems good to me.

**Pull request merged**